Add Helm-Operator migration docs (#2235)
DataDog/helm-charts: charts/datadog/docs/Migration_Helm_to_Operator.md

> # Datadog Helm chart to Datadog Operator Migration Guide

Reviewer: ```suggestion
# Migrating from Helm to Datadog Operator
```

## Overview

This guide breaks down the process for migrating from the Datadog Helm chart to the Datadog Operator for managing the Datadog Agent in Kubernetes. Using the Datadog Operator offers the following advantages:

* Operator configuration is more flexible for future enhancements.
* Validation for your Agent configurations.
* Orchestration for creating and updating Agent resources. 
* As a Kubernetes Operator, the Datadog Operator is treated as a first-class resource by the Kubernetes API. 

> * Unlike the Helm chart, the Operator is included in the Kubernetes reconciliation loop.

Reviewer 2: maybe we could drop this output and mention: DAPs, ResourceCRDs, FA/Remote Agent Management (in the future).

Learn more about the [Datadog Operator][1] and its benefits.

## Prerequisites

* Datadog Helm chart version X.X.X+
* Helm version 3.17.0+

> ## Migrate Existing Datadog Helm Release

Reviewer: ```suggestion
## Migrate existing Datadog Helm release
```

To migrate Datadog Agent workloads deployed by an existing Datadog Helm release to the DatadogAgent custom resource definition, use the built-in migration tooling available in Datadog Helm chart version X.X.X and later:

1. Configure `datadog-values.yaml`
    
    Add the following to your `datadog-values.yaml` file:

    ```yaml
   datadog:
      operator:
         enabled: true
         migration:

>          enabled: true

Reviewer 2: ```suggestion
         migration:
           enabled: true
```

   
   operator:

>       datadogCRDs:

Reviewer 3: should we specify a minimum operator version here?

         keepCrds: true
    ```

2. Upgrade your Helm release with the above configuration file

   Run:

   ```shell
   helm upgrade <RELEASE_NAME> -f datadog-values.yaml datadog/datadog
   ```

3. Confirm Agent installation

   Verify that Agent pods (tagged with `app.kubernetes.io/component:agent` and `app.kubernetes.io/managed-by: datadog-operator`) are updating according to the configured update strategy and reporting on the Containers page in Datadog. Agent pods are detected within a few minutes of deployment.

   Your Datadog Agent workloads are now managed by the DatadogAgent custom resource. To view and save the migrated DatadogAgent custom resource, run: 

   ```shell
   kubectl get datadogagents
   NAME      AGENT              CLUSTER-AGENT         CLUSTER-CHECKS-RUNNER   AGE
   datadog   Updating (5/0/0)   Progressing (1/0/1)                           5s

   kubectl get datadogagent datadog -oyaml > datadog.yaml
   ```

> For more advanced configurations and use-cases, refer to [Datadog Helm chart to Datadog Operator Advanced Migration Guide][2].

Reviewer: reminder: If you want to only publish this first guide, then this line should be removed

## Uninstall Datadog Helm chart

After migrating your Datadog Agent workloads and validating that the Agent pods are reporting as expected, you can now uninstall the Datadog Helm chart in preparation for installing the Datadog Operator.

1. Run:

   ```shell
   helm uninstall <RELEASE_NAME>
   ```

Datadog Agent pods should remain unaffected and Datadog custom resource definitions (CRDs) should remain installed on the Kubernetes cluster. 

## Install Datadog Operator Helm chart

> 

Reviewer: ```suggestion
After you uninstall the Datadog Helm chart, install the Datadog Operator Helm chart.
```

1. Run:

   ```shell
   helm install <RELEASE_NAME> \
      --set apiKeyExistingSecret=datadog-secret \
      --set appKeyExistingSecret=datadog-secret \
      --take-ownership \
      datadog/datadog-operator
   ```

2. Confirm Datadog Operator installation

>    Verify that Datadog Operator pod is reporting on the Containers page in Datadog. 

Reviewer: ```suggestion
2. Verify that Datadog Operator pod is reporting on the Containers page in Datadog. 
```

We could potentially also add a link to `https://app.datadoghq.com/containers` here.

To customize the Operator configuration, create a values.yaml file that can override the default Datadog Operator Helm chart [values][3]. 

## DatadogAgent custom resource configuration

> Now, you can manage your Datadog Agent workloads using the DatadogAgent custom resource. To make updates to your Datadog Agent deployment, modify the configuration file containing your DatadogAgent spec and deploy it on your cluster:

Reviewer: ```suggestion
After you install the Datadog Operator Helm chart, you can manage your Datadog Agent workloads using the DatadogAgent custom resource. To make updates to your Datadog Agent deployment, modify the configuration file containing your DatadogAgent spec and deploy it on your cluster:
```

```shell
kubectl apply -f datadog.yaml
```

> For a full list of configuration options, see the DatadogAgent configuration [spec][4].

Reviewer: ```suggestion
For a full list of configuration options, see the [DatadogAgent configuration spec][4].
```

Expanding the link text to be a more complete title — this helps with accessibility / screenreaders and potentially AI web crawlers

## Prerequisites

* Helm version 3.17.0+
* Datadog Helm chart version X.X.X+
* Datadog Operator Helm chart version 2.17.0+
* Datadog Operator v1.22.0+

## Migrate existing Datadog Helm release

To migrate Datadog Agent workloads deployed by an existing Datadog Helm release to the DatadogAgent custom resource definition, use the built-in migration tooling available in Datadog Helm chart version X.X.X and Datadog Operator version 1.22.0 and later.

The migration tooling supports the following Datadog Helm chart configuration options either minimally or partially:

* Agent credentials and Kubernetes secrets
* Cluster and Datadog site settings
* Tags, environment variables, and name overrides
* Pod-level overrides (partial)
* Kubelet and container runtime sockets
* Network policy (basic)
* APM (hostPort/UDS modes) with basic instrumentation and error tracking
* Logs collection (basic)
* Process monitoring
* DogStatsD
* Cluster Agent overrides and Admission Controller
* Cluster checks and cluster checks runner
* Kubernetes events, KSM core check (basic), Orchestrator Explorer (basic), and Helm check
* Prometheus scraping (partial)
* Remote Configuration

> Support for additional Datadog Helm chart configurations will be expanded in upcoming releases.

Reviewer: We have to remove this line — our documentation should never make promises about things that can happen in the future
